src/freertos_drivers/esp32/Esp32WiFiManager.hxx

```diff
@@ -85,11 +87,42 @@ public:
     /// @param cfg is the WiFiConfiguration instance used for this node. This
     /// will be monitored for changes and the WiFi behavior altered
     /// accordingly.
+    /// @param hostname_prefix is the hostname prefix to use for this node.
+    /// The @ref NodeID will be appended to this value. The maximum length for
+    /// final hostname is 32 bytes.
+    /// @param wifi_mode is the WiFi operating mode. When set to WIFI_MODE_STA
+    /// the Esp32WiFiManager will attempt to connect to the provided WiFi SSID.
+    /// When the wifi_mode is WIFI_MODE_AP the Esp32WiFiManager will create an
+    /// AP with the provided SSID and PASSWORD. When the wifi_mode is
+    /// WIFI_MODE_APSTA the Esp32WiFiManager will connect to the provided WiFi
+    /// AP and create an AP with the SSID of "<hostname>" and the provided
+    /// password. Note, the password for the AP will not be used if
+    /// soft_ap_auth is set to WIFI_AUTH_OPEN (default).
+    /// @param station_static_ip is the static IP configuration to use for the
+    /// Station WiFi connection. If not specified DHCP will be used instead.
+    /// @param primary_dns_server is the primary DNS server to use when a
+    /// static IP address is being used. If left as the default (ip_addr_any)
+    /// the Esp32WiFiManager will use 8.8.8.8 if using a static IP address.
+    /// @param soft_ap_channel is the WiFi channel to use for the SoftAP.
+    /// @param soft_ap_max_clients is the maximum number of stations that can
+    /// connect to the SoftAP. This is limited to 1-4.
+    /// @param soft_ap_auth is the authentication mode for the AP when
+    /// wifi_mode is set to WIFI_MODE_AP or WIFI_MODE_APSTA.
+    /// @param softap_static_ip is the static IP configuration for the SoftAP,
+    /// when not specified the SoftAP will have an IP address of 192.168.4.1.
     ///
     /// Note: Both ssid and password must remain in memory for the duration of
     /// node uptime.
     Esp32WiFiManager(const char *ssid, const char *password,
-        openlcb::SimpleCanStack *stack, const WiFiConfiguration &cfg);
+        openlcb::SimpleCanStack *stack, const WiFiConfiguration &cfg,
+        const char *hostname_prefix = "esp32_",
+        wifi_mode_t wifi_mode = WIFI_MODE_STA,
+        tcpip_adapter_ip_info_t *station_static_ip = nullptr,
+        ip_addr_t primary_dns_server = ip_addr_any,
+        uint8_t soft_ap_channel = 1,
+        uint8_t soft_ap_max_stations = 4,
```

Comment:
Likely not, at least I can't think of anything specific. On the CS this is intended mainly as means for initial config but it can also be used for non-home (ie: train show) environments.

Reply:
i would recommend to remove this configuration option as it is unnecessary complexity in your code.

Reply:
removed